### PR TITLE
Add `EmptyArray` and `IsEmptyArray` types

### LINKED ISSUE #929: `EmptyArray` type
There is already [`EmptyObject`](https://github.com/sindresorhus/type-fest/blob/main/source/empty-object.d.ts), and I think a type for a strictly empty array would be useful.

### PR DESCRIPTION
**Pull request by derodero24**
Closes #929

Adds `EmptyArray` (`never[]`) for representing a strictly empty array, analogous to `EmptyObject`. Also adds `IsEmptyArray<T>` as a type guard.

`IsEmptyArray` handles:
- `[]`, `never[]`, `readonly []`, `readonly never[]` → `true`
- Non-empty tuples, `string[]`, `any`, `never` → `false`

This picks up where #931 left off, with the readonly array tests that were requested in review.

### CODE REVIEW

File: test-d/empty-array.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+// IsEmptyArray — Non-empty cases
+expectType<IsEmptyArray<[1]>>(false);
+expectType<IsEmptyArray<[string, number]>>(false);
+expectType<IsEmptyArray<readonly [1]>>(false);
+expectType<IsEmptyArray<string[]>>(false);
+expectType<IsEmptyArray<readonly string[]>>(false);
+
+// IsEmptyArray — Edge cases
+expectType<IsEmptyArray<any>>(false);
+expectType<IsEmptyArray<never>>(false);
```
Comment (sindresorhus) [inline test-d/empty-array.ts:45]: Some more tests:


```suggestion
expectType<IsEmptyArray<never>>(false);

// IsEmptyArray — Union behavior
expectType<IsEmptyArray<[] | [1]>>({} as boolean);
expectType<IsEmptyArray<readonly [] | readonly [1]>>({} as boolean);

// IsEmptyArray — More non-empty/array boundaries
expectType<IsEmptyArray<[never]>>(false);
expectType<IsEmptyArray<unknown[]>>(false);
expectType<IsEmptyArray<any[]>>(false);
```

File: source/empty-array.d.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+const defaultItems: EmptyArray = [];
+
+// Prevents accidental mutation
+// @ts-expect-error
+defaultItems.push(1);
+```
+
+@category Array
+*/
+export type EmptyArray = never[];
```
Comment (sindresorhus) [inline source/empty-array.d.ts]: Why not 
```suggestion
export type EmptyArray = [];
```

?

File: test-d/empty-array.ts (hunk 1 above)
Comment (sindresorhus) [reply test-d/empty-array.ts:45]: Maybe also:

```ts
// @ts-expect-error
emptyArray.length = 1;
```

File: source/empty-array.d.ts (hunk 2 above)
Comment (derodero24) [reply source/empty-array.d.ts]: All addressed — simplified to `[]`, added union/boundary tests, and the `length` immutability test.
